tests/ui/mir/static-by-value-dyn.rs

```diff
@@ -0,0 +1,14 @@
+//! Regression test for #121176
+//! KnownPanicsLint used to assert ABI compatibility in the interpreter,
+//! which ICEs with unsized statics.
+//@ needs-rustc-debug-assertions
+
+use std::fmt::Debug;
+
+static STATIC_1: dyn Debug + Sync = *();
```

Comment:
For instance, this is a clearly ill-formed static. Nothing should ever look at its MIR. Trying to make the MIR interpreter resistant against bogus MIR is a pointless game of whack-a-mole.

Reply:
Well, we can check the tail manually for slices and dyn trait

Reply:
But that requires unfolding the type which will trigger the same cycle error, won't it?